docs/advanced-topics/serialization/inetworkserializable.md

```diff
@@ -73,32 +72,114 @@ The below `INetworkSerializable` implementation example works only with RPCs and
 [Read More About Custom NetworkVariable Implementations](../../basics/networkvariable.md)
 :::
 
+Because of the limitations of NetworkVariable, which is implemented to avoid creating garbage as much as possible, C# arrays aren't supported in NetworkVariable, including arrays that are embedded in structs, as they prevent those structs from being allocated without creating garbage. Unity provides alternatives to C# arrays, however, that do work with NetworkVariable - `NativeArray<>` and `NativeList<>`.
+
+`NativeArray<>` works similarly to C# arrays in that they are statically sized, with the size determined at allocation, while `NativeList<>` behaves similarly to `List<>` in being dynamically resizable. `NativeList<>` requires importing the `collections` package, but it does have one advantage over `NativeArray<>` in that it has a return-by-ref `ElementAt()` method. Since the `BufferSerializer` API is bi-directional serialization that takes values by ref for both reading and writing, this method simplifies serialization of array values and avoids the need to copy to and from temporary values.
```

Comment:
I would remove this whole paragraph. They can/should go to the documentation about these types to learn about them, it's an implementation detail in the context of a document about INetworkSerializable